### ISSUE TITLE
Elements Panel steals keyboard arrow events from the comments pane

### ISSUE THREAD

**Issue by jasonLaster** [RecordReplay/devtools#4578]
https://app.replay.io/recording/44662e48-c367-49ff-8780-1a1f8c362d9d

**Comment by jasonLaster** [comment]
Left a bunch of comments - 

1. feels related to this PR     `Call preventDefault on cursor key events in the markup panel` (#4423)
2. i dont trust our breakpoint locations there are a bunch of spots in KeyShortcuts that feel like they should be hit

**Comment by hbenl** [comment]
It's not the Elements panel that's doing the stealing but the ReactDevTools panel, see https://app.replay.io/recording/5895bcc2-c7d2-47c5-8456-597198e3993d
We probably didn't notice this before because the global keydown event listener is only registered when ReactDevTools is rendered and in the past we only rendered it when it was visible. But since #4537 we're always rendering it (even if it's disabled in the user's settings).
We could fix this by asking the ReactDevTools developers to not register their event listener on the document node or by putting ReactDevTools in an iframe.

**Comment by jasonLaster** [comment]
Thanks. It sounds like scoping the event listener is a pretty good short-term fix, but we might have other collisions in the future as well.

**Comment by jasonLaster** [comment]
It looks like [views/DevTools.js](https://github.com/facebook/react/blob/main/packages/react-devtools-shared/src/devtools/views/DevTools.js#L239) is adding a global keydown handler. We can either fix this by running React DevTools in an iframe or scoping this handler to React DevTools root node.

cc @bvaughn

**Comment by jcmorrow** [comment]
Here's a comment in React Devtools mentioning the issue, might be a good intervention point: https://github.com/facebook/react/blob/8b8817845a8efd8cf53be395fafbb1d0f62c4368/packages/react-devtools-shared/src/devtools/views/Components/Tree.js#L129-L131

**Comment by jasonLaster** [comment]
This feels more like an important issue than a project to me. Glad we're now triaging these in the weekly frontend meeting

**Comment by bvaughn** [comment]
Happy to lend a hand here in a week or two (if not sooner) :)

**Comment by jasonLaster** [comment]
@bvaughn looks good!

**Comment by bvaughn** [comment]
Quick update that [4.24.3](https://github.com/facebook/react/blob/main/packages/react-devtools/CHANGELOG.md#4243) was published to NPM today with my fix for this.

**Comment by bvaughn** [comment]
Okay, good news is that bumping `react-devtools-inline`, `react`, and `react-dom` packages seems doable _but_ introduces a store/bridge protocol incompatibility issue that we'll need to figure out how to solve. Need to give this a bit of thought...

**Comment by bvaughn** [comment]
I don't think I've been a part of those discussions. Maybe that would be useful? Maybe we could store some bit of information in the Replay profile that specifies which protocol version it uses. The frontend could then use this to decide how it should parse the operations array.

**Comment by abstractalgo** [comment]
Indeed, I should've been more clear, I've edited the comment (after the fact) to mention "internal discussions", sorry.

It would be awesome if we could make it work with older, newer (and even future) versions, we just haven't yet invested in fully investigating how plausible and/or easy would that be, but we were aware of at least the potential risk it might imply. Great to hear that it might be possible, and sounds like a front-end change only (i.e. the frontend part of the bridge) which is even better. :)

**Comment by bvaughn** [comment]
I've started the release process for gecko-dev this morning. Assuming that finishes without any problems, this keyboard arrow bug should be fixed in newer Replay sessions.

If we decide it needs to be fixed in older sessions too, I can do a patch fix on top of `react-devtools-inline` 4.17.0 that fixes it there too (but I assume this is not necessary).
